rabbitmq/check.py

```diff
@@ -293,6 +299,29 @@ def _get_metrics(self, data, object_type):
                     self.log.debug("Caught ValueError for %s %s = %s  with tags: %s" % (
                         METRIC_SUFFIX[object_type], attribute, value, tags))
 
+    def get_connections_stat(self, instance, base_url, object_type, vhosts=None, auth=None, ssl_verify=True, skip_proxy=False):
+        """
+        Collect metrics on currently open connection per vhost.
+        """
+        instance_proxy = self.get_instance_proxy(instance, base_url)
+        data = self._get_data(urlparse.urljoin(base_url, object_type), auth=auth,
+                              ssl_verify=ssl_verify, proxies=instance_proxy)
+
+        stats = defaultdict(int)
+        for conn in data:
+            if vhosts and conn['vhost'] in vhosts:
+                stats[conn['vhost']] += 1
+            elif not vhosts:
+                stats[conn['vhost']] += 1
+
+        # avoid having no data if no client it connected
+        if len(stats) == 0:
+            self.gauge('rabbitmq.connections', 0, tags=[])
```

Comment:
Very good point. I'll remove the the `0` and use the list of vhosts. But if the user don't set any `vhosts` in the configuration we won't send any data.